### PR TITLE
Cn OCR support in MacOS

### PR DESCRIPTION
**Pull request by QuincySan**
---
name: pull request
about: submit changes to the project
title: "[PR] Add and optimize Chinese OCR support"
labels: enhancement
assignees: ''

---

## description
This PR adds and optimizes Chinese OCR support in MacOS. These changes aim to expand Screen Pipe’s user base, particularly among Chinese-speaking users.

I used Cursor with o1 and Sonnet 3.5 to assist in generating parts of the code. As I am new to coding, if there are any areas that don’t align with the project’s style, please feel free to provide feedback. I’ve read the CONTRIBUTING.md. :)

**Note**: Since making these modifications, the original repository has had updates that I am currently unable to resolve or merge on my own. I would appreciate any assistance with handling these differences.

## type of change
- [ ] bug fix (non-breaking change which fixes an issue)
- [x] new feature (non-breaking change which adds functionality)
- [ ] breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] this change requires a documentation update

## how to test
To test these changes:
1. Checkout this branch: `CN_support`
2. Run `cargo build --release --features metal` to build the project for macOS.
3. If the web page cannot be opened, build the desktop app using the appropriate build instructions for your platform.
4. Open the page or window that contains Chinese content.
5. Select the specific app used to display Chinese content and press "search."
6. The recognized Chinese text will appear in the result and should work seamlessly with the LLM.

## screenshots (if applicable)
- Here is a screenshot of the Screen Pipe desktop page, showing Chinese characters successfully recognized by OCR in the search results.
iShot_2024-10-03_4 17 37

## checklist
- [ ] I read the Rust & programming best practices
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Other operating systems are not broken by my changes

## additional notes
- There may be some differences between the `main` branch and my `CN_support` branch due to recent updates to the original repository. I currently lack the ability to resolve these differences and would appreciate any assistance or guidance.

### CODE REVIEW

**Comment by louis030195** [comment]
@QuincySan can you fix conflict then i merge?

File: Cargo.toml [hunk 1]
```diff
@@ -43,5 +43,8 @@ git = "https://github.com/microsoft/vcpkg"
 rev = "2023.04.15"
 dynamic = true
 
+# [dependencies]
+axum = "0.7"  # 使用与您项目兼容的版本
```
Comment (louis030195) [inline Cargo.toml:47]: remove this plz

File: screenpipe-vision/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "screenpipe-vision"
-version = { workspace = true }
+version = "0.1.93"
```
Comment (louis030195) [inline screenpipe-vision/Cargo.toml:3]: revert this plz

File: screenpipe-vision/Cargo.toml [hunk 3]
```diff
@@ -38,6 +38,8 @@ screenpipe-integrations = { path = "../screenpipe-integrations" }
 tracing-subscriber = { workspace = true }
 tracing = { workspace = true }
 
+axum = "0.7"
```
Comment (louis030195) [inline screenpipe-vision/Cargo.toml:41]: remove

**Comment by QuincySan** [comment]
@louis030195 Appreciate your assistance! I have made the modifications as per your suggestions:

1. Removed the axum dependency from `Cargo.toml`
2. Restored the version number in `screenpipe-vision/Cargo.toml`
3. Removed the axum dependency from `screenpipe-vision/Cargo.toml`
4. Deleted the `screenpipe-app-tauri/src-tauri/gen/schemas/acl-manifests.json` file

Please review again. If there are any further issues or changes needed, please let me know. Thank you!

**Comment by QuincySan** [comment]
@louis030195 Thank you so much for your generous tip and recognition of my work! I truly appreciate it. 

I want to be transparent - due to regional restrictions, I may face difficulties receiving the payment through Stripe. However, please know that your appreciation itself means a lot to me.

**Pull request merged**